docs/spec/spec.md

```diff
@@ -0,0 +1,173 @@
+
+If you have any feedback or suggestions about the library, start a discussion via a [GitHub issue](https://github.com/ballerina-platform/ballerina-library/issues) or in the [Discord server](https://discord.gg/ballerinalang). Based on the outcome of the discussion, the specification and implementation can be updated. Community feedback is always welcome. Any accepted proposal, which affects the specification is stored under `/docs/proposals`. Proposals under discussion can be found with the label `type/proposal` in GitHub.
+
+The conforming implementation of the specification is released and included in the distribution. Any deviation from the specification is considered a bug.
+
+## Contents
+
+1. [Overview](#1-overview)
+2. [LDAP client](#2-ldap-client)
+    * 2.1 [The `init` method](#21-the-init-method)
+3. [Operation types](#3-operation-types)
+    * 3.1 [Add operation](#31-add-operation)
+    * 3.2 [Modify operation](#32-modify-operation)
+    * 3.3 [ModifyDN operation](#33-modifydn-operation)
+    * 3.4 [Compare operation](#34-compare-operation)
+    * 3.5 [Search operation](#35-search-operation)
+    * 3.6 [Search with type operation](#36-search-with-type-operation)
+    * 3.7 [Delete operation](#37-delete-operation)
+    * 3.8 [Close operation](#38-close-operation)
+4. [The `ldap:LdapResponse` type](#4-the-ldapldapresponse-type)
+5. [The `ldap:Error` type](#5-the-ldaperror-type)
+
+## 1. Overview
+
+The Ballerina LDAP module provides support for interacting with directory servers which support LDAP protocol. This module provided various directory operations by establishing a connection to an LDAP server, allowing for tasks such as adding, modifying, deleting, and searching directory entries. It focus on ease of use and integration, making it straightforward for developers to perform directory-based operations within their Ballerina applications.
+
+## 2. LDAP client
+
+The `ldap:Client` instance needs to be initialized before performing the functionalities. Once initialized, it can perform various operations on directories. Currently, it supports the generic LDAP operations; `add`, `modify`, `modifyDN`, `compare`, `search`, `searchWithType`, `delete`, and `close`.
+
+### 2.1 The `init` method
+
+The `init` method initializes the `ldap:Client` instance using the parameters `hostName`, `port`, `domainName`, and `password`. The `hostName` and `port` parameters are used to bind the request and authenticate clients with the directory server, while the `domainName` and `password` parameters establish the connection to the server for performing LDAP operations. In case of failure, the method returns an `ldap:Error`."
+
+```ballerina
+ldap:Client ldapClient = check new ({
+   hostName,
+   port,
+   domainName,
+   password
+});
+```
+
+## 3. Operation types
+
+The currently supported operation types in LDAP are listed here.
+
+### 3.1 Add operation
```

Comment:
API spec is not the same as API documentation. API specification deals with what a developer can do if he wants to write a specific library. So, the API spec would provide the relevant configurations types, request/response data types, and client API with proper documentation. 

IMO what we hv written is a user guide where users can refer to when using the LDAP client.

Reply:
LDAP already has their own spec explaining all the operations. I thought it would be redundant to add all those information here as well. I only added the parts that is relevant to the Ballerina LDAP client here

Reply:
LDAP can have their own spec, but Ballerina LDAP should have a spec which describes how a developer can implement a LDAP package by him self.